Content.Shared/Body/Systems/Body/SharedBodySystem.Managerial.cs

```diff
@@ -0,0 +1,305 @@
+
+        if (!slot.ContainerSlot.Insert(part))
+            return false;
+
+        var ev = new PartAddedToBodyEvent(body.Owner, part, slot.Id);
+        RaiseLocalEvent(body.Owner, ev);
+        RaiseLocalEvent(part, ev);
+
+        return true;
+    }
+
+    #endregion
+
+    #region Removing Parts
+
+    /// <summary>
+    /// Checks if a body part can be removed from the provided slot
+    /// </summary>
+    public bool CanRemove(EntityUid uid, BodyPartSlot slot, SharedBodyComponent? body = null)
+    {
+        if (!Resolve(uid, ref body))
+            return false;
+
+        if (slot.Part == null)
+            return false;
+
+        return slot.ContainerSlot?.CanRemove(slot.Part.Value, EntityManager) ?? false;
+    }
+
+    /// <summary>
+    /// Tries to remove the body part in the provided slot
+    /// </summary>
+    /// <returns></returns>
+    public bool TryRemovePart(EntityUid uid, BodyPartSlot slot, [NotNullWhen(true)] out Dictionary<BodyPartSlot, SharedBodyPartComponent>? dropped,
+        SharedBodyComponent? body = null)
+    {
+        dropped = null;
+        if (!Resolve(uid, ref body))
+            return false;
+
+        if (!CanRemove(uid, slot, body))
+            return false;
+
+        if (!body.Slots.TryGetValue(slot.Id, out var ownedSlot) ||
+            ownedSlot != slot)
+            return false;
+
+        var oldPart = slot.Part;
+        if (!RemovePart(uid, slot, body))
+            return false;
+
+        dropped = GetHangingParts(uid, slot, body);
+
+        if (oldPart != null && TryComp<SharedBodyPartComponent>(oldPart.Value, out var comp))
+            dropped[slot] = comp;
+
+        return true;
+    }
+
+    /// <summary>
+    ///     Removes the given part from the body. Defaults to finding the slot and removing the part that way.
+    /// </summary>
+    public bool RemovePart(EntityUid uid, SharedBodyPartComponent part,
+        SharedBodyComponent? body = null)
+    {
+        if (!Resolve(uid, ref body))
+            return false;
+
+        foreach (var (_, slot) in body.Slots)
+        {
+            if (slot.HasPart && slot.Part == part.Owner)
+                return RemovePart(uid, slot, body);
+        }
+
+        return false;
+    }
+
+    /// <summary>
+    ///     Removes the part inside of a body part slot.
+    /// </summary>
+    public bool RemovePart(EntityUid uid, BodyPartSlot slot,
+        SharedBodyComponent? body = null)
+    {
+        if (!Resolve(uid, ref body))
+            return false;
+
+        if (slot.Part == null) return false;
+
+        var old = slot.Part;
+
+        if (!RemovePartAndRaiseEvents(old.Value, slot, body))
+            return false;
+
+        foreach (var hangingPart in GetHangingParts(uid, slot, body))
+        {
+            RemovePart(uid, hangingPart.Key, body);
+        }
+
+        return true;
+    }
+
+    protected bool RemovePartAndRaiseEvents(EntityUid part, BodyPartSlot slot, SharedBodyComponent body)
+    {
```

Comment:
can this (and the counterpart `AddPartAndRaiseEvents` just get inlined into their respective functions? afaict these are only called in one place internally and it's clearer that way